src/routes/v2/authenticatedSites/media.ts

```diff
@@ -445,6 +447,8 @@ export class MediaRouter {
       attachRollbackRouteHandlerWrapper(this.deleteMediaFile)
     )
 
+    router.use(catchNonExistentRoutesMiddleware)
+
```

Comment:
why not put this as the terminating middleware in our `server.js`? this would allow us to catch all missing routes and to return a sensible response

Reply:
a terminating middleware already exists at `server.js` where our overall error handler resides. 

the problem exists when a subrouter captures a group and does not handle it, causing a timeout. 
do note that some of our routes fall through while others dont, so i couldnt append this at the end of all subrouters because that broke quite a number of functionality within the cms, and the manual test space for such a pr will be unreasonably large. stance here will be to monitor for 5xx errors and then fix them

Reply:
actually that's weird because i always thought if the subrouter doesn't capture the route, it should fall over to the next one. 

ie, if subrouter A captures `/about/a` and i get a request with `/about/b`, it should fall to the next router in the stack. 

this isn't very important given that we assume most users are calling from the frontend but it is weird tho